presto-main/src/main/java/com/facebook/presto/operator/DecodedBlockNode.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.block.Block;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+class DecodedBlockNode
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(DecodedBlockNode.class).instanceSize();
+
+    private final Object decodedBlock;
+    private final List<DecodedBlockNode> children;
+
+    public DecodedBlockNode(Object decodedBlock, List<DecodedBlockNode> children)
+    {
+        this.decodedBlock = requireNonNull(decodedBlock, "decodedBlock is null");
+        this.children = requireNonNull(children, "children is null");
+    }
+
+    public Object getDecodedBlock()
+    {
+        return decodedBlock;
+    }
+
+    public List<DecodedBlockNode> getChildren()
+    {
+        return children;
+    }
+
+    public long getRetainedSizeInBytes()
+    {
+        int size = INSTANCE_SIZE;
+        if (decodedBlock instanceof Block) {
```

Comment:
What if decodedBlock is not an instance of Block? Shouldn't the memory used by that object be accounted for?

Reply:
The ColumarXXX objects were counted in each BlockEncodingBuffers. But now I think they should be counted here instead because BlockEncodingBuffers would count them repeatedly.

Reply:
@yingsu00 I'm still not seeing how the size of ColumarXXX objects is accounted for. Would you explain some more?

Reply:
It should already appear in DecodedBlockNode.getRetainedSizeInBytes()